### PR TITLE
added more examples on graphical degree sequence

### PR DESCRIPTION
**Pull request by Lukong123**

### CODE REVIEW

File: networkx/algorithms/graphical.py [hunk 1]
```diff
@@ -103,6 +103,14 @@ def is_valid_degree_sequence_havel_hakimi(deg_sequence):
     valid : bool
         True if deg_sequence is graphical and False if not.
 
+    Examples
+    --------
+    >>> G = nx.Graph()
+    >>> G.add_edges_from([(1, 2), (1, 3), (2, 3), (3, 4), (4, 2), (5, 1), (5, 4)])
```
Comment (dschult) [inline networkx/algorithms/graphical.py:109]: Can you combine these two lines?
```suggestion
    >>> G = nx.Graph([(1, 2), (1, 3), (2, 3), (3, 4), (4, 2), (5, 1), (5, 4)])
```
And similarly in other examples below.

File: networkx/algorithms/graphical.py [hunk 2]
```diff
@@ -325,6 +356,17 @@ def is_digraphical(in_sequence, out_sequence):
...
 
+    Examples
+    --------
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(1, 2), (1, 3), (2, 3), (3, 4), (4, 2), (5, 1), (5, 4)])
+    >>> in_deg = dict(G.in_degree())
+    >>> in_seq = list(in_deg.values())
+    >>> out_deg = dict(G.out_degree())
+    >>> out_seq = list(out_deg.values())
+    >>> nx.is_digraphical(in_seq, out_seq)
```
Comment (dschult) [inline networkx/algorithms/graphical.py:367]:
```suggestion
    >>> G = nx.DiGraph([(1, 2), (1, 3), (2, 3), (3, 4), (4, 2), (5, 1), (5, 4)])
    >>> nx.is_digraphical([deg for _, deg in G.in_degree], [deg for _, deg in G.out_degree])
```
Using list comprehensions on getting the degrees makes this perhaps more simple?

**Review by dschult — COMMENTED** [review]
Just a couple suggestions for more compact examples.
Thanks!

File: networkx/algorithms/graphical.py (hunk 1 above)
Comment (Lukong123) [reply networkx/algorithms/graphical.py:109]: Okay sir

File: networkx/algorithms/graphical.py (hunk 2 above)
Comment (Lukong123) [reply networkx/algorithms/graphical.py:367]: Okay sir I will simplify it.

**Comment by dschult** [comment]
Would it help to have examples where the sequences are *not* graphical sequences? 
If that is better for another PR we can leave this as it is. I think the examples here all look good.
 
For example:
```python
>>> nx.is_valid_degree_sequence_havel_hakimi([1, 2, 2])
False
```

**Comment by MridulS** [comment]
@Lukong123 just a gentle ping, let me know if you have the time to add the non graphical sequence examples in this PR. If not, no worries we can do a new PR with those examples :)

**Comment by Lukong123** [comment]
> @Lukong123 just a gentle ping, let me know if you have the time to add the non graphical sequence examples in this PR. If not, no worries we can do a new PR with those examples :)

Hello, please is it still open to to work on this.

**Comment by MridulS** [comment]
> Hello, please is it still open to to work on this.

Yes! Please feel free to add more examples here.

**Review by rossbar — APPROVED** [review]
IMO the addition of examples is an improvement - I'd vote to add/modify the examples in followup PRs!

**Comment by rossbar** [comment]
FWIW I pushed up a few minor fixups (var-renaming, rm extra spaces) and **rebased** on main to get the CI updated without screwing up the diff.

**Comment by dschult** [comment]
Thanks for the updates. 
To add the non-graphical examples, one way to proceed would be to add 1 to the last entry in the sequence.
That requires making `sequence` a list instead of a generator. Then adding something like this should do it.
```python
    >>> sequence[-1] += 1
    >>> nx.is_multigraphical(sequence)
    False
```

**Comment by MridulS** [comment]
@dschult Let's create a good first issue out of it :)

**Pull request merged**